src/main/scala/pricemigrationengine/services/CohortTable.scala

```diff
@@ -28,31 +32,60 @@ object CohortTable {
   def update(result: EstimationResult): ZIO[CohortTable, CohortUpdateFailure, Unit] =
     ZIO.accessM(_.get.update(result))
 
+  private implicit val cohortItemDeserialiser: DynamoDBDeserialiser[CohortItem] =
+    cohortItem =>
+      getStringFromResults(cohortItem, "subscriptionNumber")
+        .map(CohortItem.apply)
+
+  private implicit val estimationResultSerialiser: DynamoDBUpdateSerialiser[EstimationResult] =
+    estimationResult =>
+      Map(
+        stringFieldUpdate("processingStage", EstimationComplete.value),
+        dateFieldUpdate("expectedStartDate", estimationResult.expectedStartDate),
+        doubleFieldUpdate("estimatedNewPrice", estimationResult.estimatedNewPrice)
+      ).asJava
+
+  private implicit val cohortTableKeySerialiser: DynamoDBSerialiser[CohortTableKey] =
+    estimationResult =>
+      Map(stringUpdate("subscriptionNumber", estimationResult.subscriptionNumber)).asJava
+
+  private def stringFieldUpdate(fieldName: String, stringValue: String) =
+    fieldName -> new AttributeValueUpdate(new AttributeValue().withS(stringValue), AttributeAction.PUT)
+
+  private def dateFieldUpdate(fieldName: String, dateValue: LocalDate) =
+    fieldName -> new AttributeValueUpdate(new AttributeValue().withS(dateValue.format(DateTimeFormatter.ISO_LOCAL_DATE)), AttributeAction.PUT)
+
+  private def doubleFieldUpdate(fieldName: String, doubleValue: Double) =
+    fieldName -> new AttributeValueUpdate(new AttributeValue().withS(doubleValue.toString), AttributeAction.PUT)
+
+  private def stringUpdate(fieldName: String, stringValue: String) =
+    fieldName -> new AttributeValue().withS(stringValue)
+
   val impl: ZLayer[DynamoDBZIO, Nothing, UserRepo] =
-    ZLayer.fromFunction { dynamoDBZStream: DynamoDBZIO =>
+    ZLayer.fromFunction { dynamoDBZIO: DynamoDBZIO =>
       new Service {
         override def fetch(filter: CohortTableFilter, batchSize: Int): IO[CohortFetchFailure, ZStream[Any, CohortFetchFailure, CohortItem]] = {
           UIO(
-            dynamoDBZStream.get.query(
-              new QueryRequest()
-                .withTableName("PriceMigrationEngineDev")
-                .withIndexName("ProcessingStageIndex")
-                .withKeyConditionExpression("processingStage = :processingStage")
-                .withExpressionAttributeValues(
-                  Map(":processingStage" -> new AttributeValue(filter.value)).asJava
-                )
-                .withLimit(batchSize),
-              { result =>
-                getStringFromResults(result, "subscriptionNumber").map(CohortItem.apply)
-              }
-            ).mapError(error => CohortFetchFailure(error.toString))
+            dynamoDBZIO
+              .get
+              .query(
+                new QueryRequest()
+                  .withTableName("PriceMigrationEngineDev")
```

Comment:
Might be easier if this name is configurable, as will probably have a different table for each cohort.

Reply:
Ill create a separate PR with the changes to add the configuration.

Reply:
👍